quickjs.c

```diff
@@ -10083,6 +10085,19 @@ void JS_SetOpaque(JSValue obj, void *opaque)
    JSObject *p;
     if (JS_VALUE_GET_TAG(obj) == JS_TAG_OBJECT) {
         p = JS_VALUE_GET_OBJ(obj);
+        if (p->class_id >= JS_CLASS_INIT_COUNT)
+            p->u.opaque = opaque;
+        // TODO(saghul) report error?
+    }
+}
+
+/* User code can't set the opaque of internal objects. */
+static void JS_SetOpaqueInternal(JSValue obj, void *opaque)
+{
+   JSObject *p;
+    if (JS_VALUE_GET_TAG(obj) == JS_TAG_OBJECT) {
```

Comment:
Is this "is object?" check necessary for internal code? It's almost certainly a bug if we're calling it on a non-object, right?

Reply:
Yep. I'll add an assert instead!